File: fixture-monkey-tests/java-concurrent-tests/src/test/java/com/navercorp/fixturemonkey/tests/concurrent/ConcurrentTest.java
package com.navercorp.fixturemonkey.tests.concurrent;

import static com.navercorp.fixturemonkey.tests.TestEnvironment.TEST_COUNT;
import static org.assertj.core.api.BDDAssertions.then;

import java.util.Map;

import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

import com.navercorp.fixturemonkey.FixtureMonkey;
import com.navercorp.fixturemonkey.api.introspector.ConstructorPropertiesArbitraryIntrospector;

@Disabled("It will be enabled when parallelism is supported")
class ConcurrentTest {
	private static final FixtureMonkey SUT = FixtureMonkey.builder()
		.objectIntrospector(ConstructorPropertiesArbitraryIntrospector.INSTANCE)
		.build();

	@Test
	void test1() {
		JavaObject actual = SUT.giveMeOne(JavaObject.class);

		then(actual).isNotNull();
	}

	@Test
	void test2() {
		JavaObject actual = SUT.giveMeOne(JavaObject.class);

		then(actual).isNotNull();
	}

	@Test
	void test3() {
		JavaObject actual = SUT.giveMeOne(JavaObject.class);

		then(actual).isNotNull();
	}

	public record JavaObject(
		String value,
		Map<String, String> map
	) {

	}
}

File: fixture-monkey-tests/kotlin-concurrent-tests/src/test/kotlin/com/navercorp/fixturemonkey/tests/concurrent/kotlin/ConcurrentTest.kt
package com.navercorp.fixturemonkey.tests.concurrent.kotlin

import com.navercorp.fixturemonkey.FixtureMonkey
import com.navercorp.fixturemonkey.kotlin.KotlinPlugin
import com.navercorp.fixturemonkey.kotlin.giveMeOne
import com.navercorp.fixturemonkey.tests.TestEnvironment
import org.assertj.core.api.BDDAssertions.then
import org.junit.jupiter.api.RepeatedTest

class ConcurrentTest {
    @RepeatedTest(TestEnvironment.TEST_COUNT)
    fun test1() {
        val actual: KotlinObject = SUT.giveMeOne()
        then(actual).isNotNull
    }

    @RepeatedTest(TestEnvironment.TEST_COUNT)
    fun test2() {
        val actual: KotlinObject = SUT.giveMeOne()
        then(actual).isNotNull
    }

    @RepeatedTest(TestEnvironment.TEST_COUNT)
    fun test3() {
        val actual: KotlinObject = SUT.giveMeOne()
        then(actual).isNotNull
    }

    data class KotlinObject(val value: String, val map: Map<String, String>)

    companion object {
        private val SUT = FixtureMonkey.builder()
            .plugin(KotlinPlugin())
            .build()
    }
}
